gson/src/main/java/com/google/gson/GsonBuilder.java

```diff
@@ -362,15 +362,15 @@ public GsonBuilder setPrettyPrinting() {
   }
 
   /**
-   * By default, Gson is strict and only accepts JSON as specified by
-   * <a href="http://www.ietf.org/rfc/rfc4627.txt">RFC 4627</a>. This option makes the parser
-   * liberal in what it accepts.
+   * By default, Gson is liberal in what it accepts. Setting lenient to false causes Gson to
+   * only accept JSON as specified by 
+   * <a href="http://www.ietf.org/rfc/rfc4627.txt">RFC 4627</a>.
    *
    * @return a reference to this {@code GsonBuilder} object to fulfill the "Builder" pattern
    * @see JsonReader#setLenient(boolean)
    */
-  public GsonBuilder setLenient() {
-    lenient = true;
+  public GsonBuilder setLenient(boolean lenient) {
```

Comment:
In order to maintain compatibility, we could add a `setLenient(boolean)` method to GsonBuilder in addition to keeping the `setLenient(void)` method even though calling the `setLenient(void)` method would not have much effect.  Perhaps the `setLenient(void)` method should also be marked deprecated in order to discourage usage of a method that has little effect.

I also considered adding a `setStrict(void)` method which would set the lenient flag to false instead of having the `setLenient(boolean)` method 

Would you be okay with either of these solutions?